ionc/ion_timestamp.c

```diff
@@ -134,13 +134,86 @@ iERR _ion_timestamp_initialize(ION_TIMESTAMP *ptime)
     iRETURN;
 }
 
+/**
+ * Writes the given string containing a fraction expressed in scientific notation in expanded form. For example,
+ * the string "1E-8" will be written as ".00000001".
+ * @param scientific_fraction a string representing a number in the range [0, 1).
+ * @param exponent the exponent of the number denoted by `scientific_fraction`.
+ * @param buffer the destination for the expanded fraction.
+ * @param end_of_buffer the end of `buffer`.
```

Comment:
I haven't written a lot of C recently, but I seem to recall `(char * buffer, int len)` being a more common way to represent a buffer than `(char * start, char * end)`. `ion_timestamp_to_string` uses this convention.

Either way, it's unclear from the doc comments whether 'the end of buffer' means it's the last writeable byte or if that's the `\0` terminator that can't be overwritten.